### PR TITLE
Decouple the creation of PodManager from ElasticDLJobService

### PR DESCRIPTION
**Pull request by brightcoder01**
This is the first step of PodManager refactor. There will be three in total.

- Decouple PodManager creation from ElasticDLJobService.
- Decouple PodManager with TaskManager and RendezvousServer using PodEventCallbacks.
- More Polish on the __init__ parameter building and the composition among different components.

### CODE REVIEW

File: elasticdl/python/tests/k8s_instance_manager_test.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 from unittest.mock import MagicMock, call
 
 from elasticdl.python.common.k8s_client import PodType
-from elasticdl.python.master.k8s_instance_manager import InstanceManager
+from elasticdl.python.master.pod_manager import PodManager
```
Comment (brightcoder01) [inline elasticdl/python/tests/k8s_instance_manager_test.py]: I will rename k8s_instance_manager_test to pod_manager_test in the next PR.
In the current PR, I keep the original file name to check the difference of the file content.

File: elasticdl/python/master/elasticdl_job_service.py [hunk 2]
```diff
@@ -78,52 +95,25 @@ def __init__(self, args, task_manager, rendezvous_server=None):
...
-            )
-        ):
-            job_type = JobType.PREDICTION_ONLY
-        else:
-            job_type = JobType.TRAINING_ONLY
-
-        return job_type
+                self.pod_manager.start_parameter_servers()
+            self.pod_manager.start_workers()
+            self.pod_manager.update_status(InstanceManagerStatus.RUNNING)
```
Comment (workingloong) [inline elasticdl/python/master/elasticdl_job_service.py:110]: Can we move the pod_manager instance out of elasticdl job service later?

File: elasticdl/python/master/master.py [hunk 3]
```diff
@@ -85,8 +104,12 @@ def stop(self):
         logger.info("Master stopped")
 
     def create_pod_manager_if_needed(self, args):
-        # TODO: set None if args.need_pod_manager is False.
-        self.pod_manager = PodManager(args)
+        if args.need_task_manager:
```
Comment (workingloong) [inline elasticdl/python/master/master.py]: Is it `args.need_pod_manager` not `args.need_task_manager`?

**Comment by workingloong** [comment]
Should we delete `k8s_instance_manager.py` ?

**Comment by brightcoder01** [comment]
> Should we delete `k8s_instance_manager.py` ?

We can delete this file in the next clean up PR

File: elasticdl/python/master/master.py (hunk 3 above)
Comment (brightcoder01) [reply elasticdl/python/master/master.py]: Good catch, updated.

File: elasticdl/python/master/elasticdl_job_service.py (hunk 2 above)
Comment (brightcoder01) [reply elasticdl/python/master/elasticdl_job_service.py:110]: Yes, we can.
Will move the instance out in the next PR.

File: elasticdl/python/master/k8s_instance_manager.py [hunk 4]
```diff
@@ -147,6 +206,29 @@ def __init__(
             self._num_ps, self._k8s_client.get_ps_service_address
         )
         self._worker_addrs = []
+        self._pod_event_callbacks = []
+        self._worker_command = None
+        self._worker_args = None
+        self._ps_command = None
+        self._ps_args = None
+
+    def set_up(
```
Comment (workingloong) [inline elasticdl/python/master/k8s_instance_manager.py:215]: Rename `set_up` to `set_pod_command` ?

Comment (brightcoder01) [reply elasticdl/python/master/k8s_instance_manager.py:215]: I prefer `set_up`. It means the set_up action which is necessary before start&run. Maybe we will add more parameters besides command arguments in this method.

**Review by workingloong — DISMISSED** [review]

**Review by workingloong — APPROVED** [review]

**Pull request merged**